support-frontend/assets/helpers/redux/checkout/personalDetails/state.ts

```diff
@@ -0,0 +1,25 @@
+import type { UserTypeFromIdentityResponse } from 'helpers/identityApis';
+import type { Title } from 'helpers/user/details';
+import { getUser } from 'helpers/user/user';
+
+export type PersonalDetailsState = {
+	title?: Title;
+	firstName: string;
+	lastName: string;
+	email: string;
+	confirmEmail?: string;
+	isSignedIn: boolean;
+	userTypeFromIdentityResponse: UserTypeFromIdentityResponse;
+	telephone?: string;
+};
+
+const user = getUser();
```

Comment:
By defining `user` here we'll have a side affect that `getUser()` is called when this module is imported. Could we convert `initialPersonalDetailsState` from na object to a function that returns an object, we can then define `user` within this function, so `getUser()` is only called at that point it's needed?

Reply:
No, the initial state has to be an object literal in order to work correctly with `createSlice`. Store components in general need to be static in order to be able to get the utility types for things like `dispatch`.

We really ought to be providing this 'starter' information after store initialisation, but unfortunately due to the way we've been using Redux up until this point we have far too much code that relies on the store having a bunch of initial state in it at the point of component render. This is something we can't really fix until everything else has been migrated.

Reply:
OK, sounds unavoidable in this case, thanks for explaining